main.tf

```diff
@@ -76,17 +87,88 @@ resource "aws_s3_bucket" "aws_logs" {
   }
 }
 
+#
+# IAM
+#
+
+data "aws_iam_policy_document" "cloudtrail_assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals = {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "cloudtrail_cloudwarch_logs" {
+  statement {
+    sid = "WriteCloudWatchLogs"
+
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    resources = ["arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:log-group:${var.cloudtrail_cloudwatch_logs_group}:*"]
+  }
+}
+
+resource "aws_iam_role" "main" {
+  count              = "${var.enable_cloudtrail ? 1 : 0}"
+  name               = "cloudtrail-cloudwatch-logs-role"
+  assume_role_policy = "${data.aws_iam_policy_document.cloudtrail_assume_role.json}"
+}
+
+resource "aws_iam_policy" "main" {
+  count  = "${var.enable_cloudtrail ? 1 : 0}"
+  name   = "cloudtrail-cloudwatch-logs-policy"
+  policy = "${data.aws_iam_policy_document.cloudtrail_cloudwarch_logs.json}"
+}
+
+resource "aws_iam_policy_attachment" "main" {
+  count      = "${var.enable_cloudtrail ? 1 : 0}"
+  name       = "cloudtrail-cloudwatch-logs-policy-attachment"
+  policy_arn = "${aws_iam_policy.main.arn}"
+  roles      = ["${aws_iam_role.main.name}"]
+}
+
+#
+# CloudWatch Logs
+#
+
+resource "aws_cloudwatch_log_group" "main" {
+  count             = "${var.enable_cloudtrail ? 1 : 0}"
+  name              = "${var.cloudtrail_cloudwatch_logs_group}"
+  retention_in_days = "${var.expiration}"
+}
+
+#
+# CloudTrail
+#
+
 resource "aws_cloudtrail" "cloudtrail" {
+  depends_on = [
```

Comment:
There's a `depends_on` at the bottom of this resource as well.

Reply:
fixed